DDG4/python/DDSim/DD4hepSimulation.py

```diff
@@ -71,14 +71,13 @@ def __init__(self):
 
     self.numberOfEvents = 0
     self.skipNEvents = 0
+    self.numberOfThreads = 1
     self.physicsList = None  # deprecated use physics.list
     self.crossingAngleBoost = 0.0
     self.macroFile = ''
     self.enableGun = False
     self.enableG4GPS = False
     self.enableG4Gun = False
-    self._g4gun = None
-    self._g4gps = None
```

Comment:
TODO: Geant4 gun and GPS event generators are not initialized correctly due to this removal.

Reply:
Despite modified implementation, this remain a TODO since only one thread reads the `macroFile`... (as of da36a3e368bbe0651fdb921a67512cb287fa438e). And Geant4GeneratorWrapper doesn't allow setting properties when shared = True.